### PR TITLE
fix(amplify-util-mock): mock AWS access key when server is running

### PR DESCRIPTION
**Pull request by cjihrig**
#### Description of changes
This commit updates function mocking to use the mock AWS access key value when the mock API server is running.

#### Issue #, if available
https://github.com/aws-amplify/amplify-cli/issues/6897

#### Description of how you validated changes
Local testing and unit tests.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ammarkarachi — APPROVED** [review]

**Review by edwardfoyle — CHANGES_REQUESTED** [review]
Unless I'm misunderstanding the linked issue, I don't think this solves the problem. The issue is that the `API_<API NAME>_GRAPHQLAPIKEYOUTPUT` env var does not get the fake API key. This is different than the `AWS_ACCESS_KEY_ID` which is modified in this PR. I think what we need is another condition [here](https://github.com/aws-amplify/amplify-cli/blob/dceb13a76a85a05940078868a3e2e1ca85656938/packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts#L64) to check for `attribute === 'GraphQLApiKeyOutput'` and if so, set the val to the mock API key

File: packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts [hunk 1]
```diff
@@ -61,9 +61,15 @@ const getAmplifyMetaParams = (
...
         print.warning('This attribute will be undefined in the mock environment until you run `amplify push`');
       }
-      if (overrideApiToLocal && attribute === 'GraphQLAPIEndpointOutput') {
-        val = `http://localhost:${MOCK_API_PORT}/graphql`;
+
+      if (overrideApiToLocal) {
+        if (attribute === 'GraphQLAPIEndpointOutput') {
+          val = `http://localhost:${MOCK_API_PORT}/graphql`;
+        } else if (attribute === 'GraphQLApiKeyOutput') {
+          val = 'da2-fakeApiId123456';
```
Comment (edwardfoyle) [inline packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts:69]: If it's not too difficult to move this value to an exported const that is imported when needed that would be nice

File: packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts [hunk 2]
```diff
@@ -61,9 +61,15 @@ const getAmplifyMetaParams = (
         );
         print.warning('This attribute will be undefined in the mock environment until you run `amplify push`');
       }
-      if (overrideApiToLocal && attribute === 'GraphQLAPIEndpointOutput') {
-        val = `http://localhost:${MOCK_API_PORT}/graphql`;
+
+      if (overrideApiToLocal) {
+        if (attribute === 'GraphQLAPIEndpointOutput') {
+          val = `http://localhost:${MOCK_API_PORT}/graphql`;
+        } else if (attribute === 'GraphQLApiKeyOutput') {
```
Comment (edwardfoyle) [inline packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts:68]: I think this should be
```suggestion
        } else if (attribute === 'GraphQLAPIKeyOutput') {
```

File: packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts [hunk 3]
```diff
@@ -61,9 +61,15 @@ const getAmplifyMetaParams = (
         );
         print.warning('This attribute will be undefined in the mock environment until you run `amplify push`');
       }
-      if (overrideApiToLocal && attribute === 'GraphQLAPIEndpointOutput') {
-        val = `http://localhost:${MOCK_API_PORT}/graphql`;
+
+      if (overrideApiToLocal) {
+        if (attribute === 'GraphQLAPIEndpointOutput') {
```
Comment (edwardfoyle) [inline packages/amplify-util-mock/src/utils/lambda/populate-cfn-params.ts:66]: nit might be a bit cleaner to make this a switch statement

File: packages/amplify-util-mock/src/api/api.ts [hunk 4]
```diff
@@ -17,6 +17,9 @@ import { getInvoker } from 'amplify-category-function';
 import { lambdaArnToConfig } from './lambda-arn-to-config';
 import { timeConstrainedInvoker } from '../func';
 
+export const GRAPHQL_API_ENDPOINT_OUTPUT = 'GraphQLAPIEndpointOutput';
+export const GRAPHQL_API_KEY_OUTPUT = 'GraphQLAPIKeyOutput';
+export const MOCK_API_KEY = 'da2-fakeApiId123456';
```
Comment (edwardfoyle) [inline packages/amplify-util-mock/src/api/api.ts:22]: although it's slightly tangental to this PR, this const should be used in the other places where we use this value as well: https://github.com/aws-amplify/amplify-cli/search?q=da2-fakeApiId123456

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**